attribute-masking-service/src/main/java/uk/gov/gchq/palisade/service/attributemask/AttributeMaskingApplication.java

```diff
@@ -84,12 +87,22 @@ public static void main(final String[] args) {
      */
     @EventListener(ApplicationReadyEvent.class)
     public void serveForever() {
-        Set<CompletableFuture<?>> runnerThreads = runners.stream()
-                .map(runner -> CompletableFuture.supplyAsync(() -> runner.run(materializer), executor))
-                .collect(Collectors.toSet());
+        runnerThreads.addAll(runners.stream()
+                .map(runner -> CompletableFuture.supplyAsync(() -> runner.run(materialiser), executor))
+                .collect(Collectors.toSet()));
         LOGGER.info("Started {} runner threads", runnerThreads.size());
-
         runnerThreads.forEach(CompletableFuture::join);
     }
 
+    /**
+     * Cancels any futures that are running and then terminates the Akka Actor so the service can be terminated safely
+     */
+    @PreDestroy
+    public void onExit() {
+        LOGGER.info("Cancelling running futures");
+        runnerThreads.forEach(thread -> thread.cancel(true));
+        LOGGER.info("Terminating actor system");
```

Comment:
do these need to be at info? wouldnt debug make more sense here?

Reply:
I think it might be prudent to show the details of what is happening when safely shutting down the services.
This is the case in the rest of the services.